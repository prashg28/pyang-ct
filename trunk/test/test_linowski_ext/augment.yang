module unique {

   namespace "urn:org:example:xml:ct:unique";
   prefix "u";

   import complex-types {
       prefix "ct"; 
   }

   ct:complex-type ct1 {
       container c {}
       leaf l {
           type int8;
       }
       choice ch {
           case a {
           }
       }
   }

   ct:complex-type ct2 {
       leaf AAA {
           type int8;
       }
   }

   ct:instance c {
       ct:instance-type ct1;

       augment "c" {
           leaf l {
               type int8;
               mandatory true;
           }

           leaf-list ll {
               type int8;
               min-elements 1;
           }

           container c1 {
               presence "PP";
               container c2 {
                   presence "PP";
                   leaf l {
                       type int8;
                       mandatory true;
                   }
               }
           }

           ct:instance ccc {
               ct:instance-type ct1;
               mandatory true;
           }

           ct:instance-list cccl {
               ct:instance-type ct1;
           }
       }

       augment "c/c" {
           ct:instance b {
               ct:instance-type ct2;
           }
       }

       augment "c/c" {
           container b { }
       }

       augment "/c" {
       }

       augment "c" {
           container c {}
       }

       augment "l" {
       }
   }


   container ac {
   }

}
