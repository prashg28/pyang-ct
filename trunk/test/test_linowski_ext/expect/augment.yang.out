augment.yang:31: error: MANDATORY_AUGMENTATION
augment.yang:36: error: MANDATORY_AUGMENTATION
augment.yang:41: error: MANDATORY_AUGMENTATION
augment.yang:52: error: MANDATORY_AUGMENTATION
augment.yang:57: error: KEY_REQUIRED
augment.yang:69: error: DUPLICATE_CHILD_NAME
augment.yang:72: error: BAD_VALUE
augment.yang:79: error: BAD_NODE_IN_AUGMENT
